samples/dashboards/sync/sync-highlight-options/demo.js

```diff
@@ -67,23 +79,68 @@ const optionsCbx = document.querySelectorAll('.option');
 
 optionsCbx.forEach(checkbox => {
     checkbox.addEventListener('change', () => {
-        const value = checkbox.dataset.value;
-        const { checked } = checkbox;
+        const { checked, dataset } = checkbox;
+        const checkboxDataValue = dataset.value;
 
-        if (value === 'enabled') {
+        if (checkboxDataValue === 'enabled') {
             optionsCbx.forEach(cbx => {
                 if (cbx.dataset.value !== 'enabled') {
                     cbx.disabled = !checked;
                 }
             });
         }
 
-        board.mountedComponents[1].component.update({
-            sync: {
-                highlight: {
-                    [value]: checked
-                }
-            }
+        const componentIndices =
+            checkbox.id === 'cbx-grid-enabled' ? [1, 2] : [0];
+        componentIndices.forEach(index => {
+            board.mountedComponents[index].component.update({
+                ...(checkboxDataValue === 'enabled' && ({
+                    sync: {
+                        highlight: {
+                            enabled: checked
+                        }
+                    }
+                })),
+                ...(checkboxDataValue === 'showTooltip' && ({
+                    chartOptions: {
+                        tooltip: {
+                            enabled: checked
+                        }
+                    }
+                })),
+                ...(checkboxDataValue === 'highlightPoint' && ({
+                    chartOptions: {
+                        plotOptions: {
+                            series: {
+                                ...(!checked && ({
+                                    events: {
+                                        mouseOver() {
+                                            return checked;
+                                        }
+                                    }
+                                })),
+                                marker: {
+                                    states: {
+                                        hover: {
+                                            enabled: checked
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                })),
+                ...(checkboxDataValue === 'showCrosshair' && ({
+                    chartOptions: {
+                        xAxis: {
+                            crosshair: checked
+                        },
+                        yAxis: {
+                            crosshair: checked
+                        }
+                    }
+                }))
+            });
```

Comment:
That's not what these options should change. What they should change, are the options that are available on [HighchartsComponentOptions.sync.highlight option](https://api.highcharts.com/dashboards/#interfaces/Dashboards_Components_HighchartsComponent_HighchartsComponentOptions.SyncOptions), only for the Highcharts Component.